common/src/main/java/com/microsoft/identity/common/internal/net/HttpRequest.java

```diff
@@ -131,13 +131,18 @@ public static HttpResponse sendPost(final URL requestUrl, final Map<String, Stri
                 new HttpStartEvent()
                         .putMethod(REQUEST_METHOD_POST)
                         .putPath(requestUrl)
-                        .putRequestIdHeader(requestHeaders.get(CLIENT_REQUEST_ID))
+                        .putRequestIdHeader(requestHeaders == null ? null : requestHeaders.get(CLIENT_REQUEST_ID))
         );
 
         final HttpRequest httpRequest = new HttpRequest(requestUrl, requestHeaders, REQUEST_METHOD_POST,
                 requestContent, requestContentType);
         final HttpResponse response = httpRequest.send();
-        Telemetry.emit(new HttpEndEvent().putStatusCode(response.getStatusCode()));
+
+        if (response != null) {
+            Telemetry.emit(new HttpEndEvent().putStatusCode(response.getStatusCode()));
```

Comment:
nit: Refactor to below for better readability. if/else seems redundant here. Feedback applies below also
```
final HttpEndEvent httpEvent = new HttpEndEvent();
if(response != null){
httpEvent.putStatusCode(response.getStatusCode());
}
Telemetry.emit(httpEvent);
```

Reply:
Addressed